docker/start_build.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+
+LOG_DIR="${HOME}/docker/logs"
+GIT_LOG="${LOG_DIR}/git.log"
+BUILD_LOG="${LOG_DIR}/build.log"
+
+cd ~/
+
+[ -n "${TAG}" ] || TAG="master"; \
+{ git pull; [ ${TAG} = 'master' ] || { git describe; git verify-tag ${TAG}; }; \
+git verify-commit ${TAG}^{commit}; git checkout --recurse-submodules ${TAG}; \
+git status; } 2>&1 | tee -a ${GIT_LOG}
+
+/home/user/derivative-maker ${@:1:$(($#-1))} 2>&1 | tee -a ${BUILD_LOG}; set -- ${@: -1}; exec "$@"
```

Comment:
Won't this cut the `current` off of `--freshness current` if the user just runs `docker/run.sh` as it currently is? Then it will try to run `current` as a command, which doesn't make much sense to me. Also the command run at the end won't be able to have any arguments passed to it. I like the idea here, but I think the feature needs fleshed out a bit more for it to be usable, perhaps with some sort of `--hook` argument for specifying a command with arguments that would be `eval`'d at the end.

For the time being, I think it would be preferable to remove the hook command feature, and add that as a separate PR once this is working well.

```suggestion
/home/user/derivative-maker "$@" | tee -a ${BUILD_LOG}
```

Reply:
Yeah I wrote this quickly, haven't tested it. There is a better way to do this for sure.